internal/controllers/machine/machine_controller.go

```diff
@@ -122,6 +126,10 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 					predicates.ResourceHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue),
 				),
 			)).
+		Watches(
```

Comment:
Took a closer look into the implementation.

I think it would be good if we also have a watch on the MDs

Example:
* Update is triggered on MD
* MD controller creates new MS with new revision
* MD controller sets the new MS revision as revision on the MD

Until this point the old MachineSet won't be changed in any way. Eventually the replicas will be decreased. But I think it would be better if we already react to the MD update where the revision on the MD Is increased (at which point the old MS is immediately outdated)